monkestation/code/game/objects/items/grenades/monkey_barrel.dm

```diff
@@ -0,0 +1,40 @@
+/obj/item/grenade/monkey_barrel
+	name = "angry monkey barrel"
+	desc = "A barrel full of monkeys ready to hand out a beating."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	item_state = "barrel"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+
+/obj/item/grenade/monkey_barrel/preprime(mob/user, delayoverride, msg = TRUE, volume = 60)
+	var/turf/T = get_turf(src)
+	log_grenade(user, T)
+	if(user)
+		add_fingerprint(user)
+		if(msg)
+			to_chat(user, "<span class='warning'>You shake the barrel, angering the monkeys inside!</span>")
+	playsound(src, 'sound/creatures/monkey/monkey_screech_7.ogg', volume, 1)
+	active = TRUE
```

Comment:
Your preprime doesn't check if it's set to active or not, so it could potentially be spammed.

Reply:
It checks if its active in the attack_self, and the original preprime doesn't check it either, so this should be fine